test/operators/dedicatedadmin.go

```diff
@@ -0,0 +1,295 @@
+	// interval is the duration in seconds between polls
+	// values here for humans
+
+	timeout := 10
+	interval := 1
+
+	// convert time.Duration type
+	timeoutDuration := time.Duration(timeout) * time.Minute
+    intervalDuration := time.Duration(interval) * time.Second
+
+	start := time.Now()
+
+	Loop:
+		for {
+			_, err = h.Kube().RbacV1().RoleBindings(projectName).Get(roleBindingName, metav1.GetOptions{})
+			elapsed := time.Now().Sub(start)
+
+			switch {
+			case err == nil:
+				log.Printf("Found rolebinding %v", roleBindingName)
+				break Loop
+			default:
+				if elapsed < timeoutDuration {
+					timeTilTimeout := timeoutDuration - elapsed
+					log.Printf("Failed to get rolebinding %v, will retry (timeout in: %v)", roleBindingName, timeTilTimeout)
+					time.Sleep(intervalDuration)
+				} else {
+					log.Printf("Failed to get rolebinding %v before timeout, failing", roleBindingName)
+					break Loop
+				}
+			}
+		}
+	return err
+}
+
+
+func pollLockFile(h *helper.H) (error) {
+	// GetConfigMap polls for a configMap with a timeout
+	// to handle the case when a new cluster is up but the OLM has not yet
+	// finished deploying the operator
+
+	var err error
+
+	// timeout is the duration in minutes that the polling should last
+	// interval is the duration in seconds between polls
+	// values here for humans
+	timeout := 20
+	interval := 5
+
+	// convert time.Duration type
+	timeoutDuration := time.Duration(timeout) * time.Minute
+    intervalDuration := time.Duration(interval) * time.Second
+
+	start := time.Now()
+
+	Loop:
+		for {
+			_, err = h.Kube().CoreV1().ConfigMaps(operatorNamespace).Get(operatorLockFile, metav1.GetOptions{})
+			elapsed := time.Now().Sub(start)
+
+			switch {
+			case err == nil:
+				// Success
+				break Loop
+			default:
+				if elapsed < timeoutDuration {
+					timeTilTimeout := timeoutDuration - elapsed
+					log.Printf("Failed to get configmap, will retry (timeout in: %v", timeTilTimeout)
+					time.Sleep(intervalDuration)
+				} else {
+					log.Printf("Failed to get configmap before timeout, failing")
+					break Loop
+				}
+			}
+		}
+
+	return err
+}
+
+
+func pollDeploymentList(h *helper.H) (*appsv1.DeploymentList, error) {
+	// pollDeploymentList polls for deployments with a timeout
+	// to handle the case when a new cluster is up but the OLM has not yet
+	// finished deploying the operator
+
+	var err error
+	var deploymentList *appsv1.DeploymentList
+
+	// timeout is the duration in minutes that the polling should last
+	// interval is the duration in seconds between polls
+	// values here for humans
+	timeout := 20
+	interval := 5
+
+	// convert time.Duration type
+	timeoutDuration := time.Duration(timeout) * time.Minute
+    intervalDuration := time.Duration(interval) * time.Second
+
+	start := time.Now()
+
+	Loop:
+		for {
+			deploymentList, err = h.Kube().AppsV1().Deployments(operatorNamespace).List(metav1.ListOptions{})
+			elapsed := time.Now().Sub(start)
+
+			switch {
```

Comment:
Agreed.  I was not happy with how that worked, but ran am unsure how I could pass in different objects since Go is so strongly typed.  

In this example: h.Kube().AppsV1().Deployments.... vs. h.Kube().CoreV1().ConfigMaps...

I need to be able to identify *what* to poll in the loop, but unsure how to pass that in, and in the Deployments example, I need to return the object, so the output would have to have a matching type.  100% this is my Go n00b though, so if you have a suggestion, let me know.  I can also ask my team leads for advice.

Reply:
I'm a super go noob. :-) If this can't be worked around, that's fine! We can resolve as is.